lua/entities/sent_tardis/init.lua

```diff
@@ -1625,6 +1625,388 @@ function ENT:TogglePower()
 	return false
 end
 
+local function CheckPP(ply, ent) -- Prop Protection
+	return hook.Call("PhysgunPickup", GAMEMODE, ply, ent)
+end
+
+local E2Commands = {
+	["Demat"] = function(self,data,pos,ang)
+		if self and IsValid(self) and CheckPP(data.player,self) then
```

Comment:
You can remove the if self and IsValid(self) from all of these (keep the CheckPP bit though on ones that have that)